### ISSUE TITLE
Add support for transforming for loops to forEach loops

### ISSUE THREAD

**Issue by aboyton** [lebab/lebab#188]
Some style guides such as [airbnb's](http://airbnb.io/javascript/#iterators--nope) prefer `forEach` loops over `for-of`.

I'd propose we make another transformation, like `for-of` that transforms for loops into `forEach` loops. Obviously the conditions of when you can do this transformation are different—you can use the index and the array, but you can't have `break` and `continue` statements.

I'm happy to help with a pull request to be able to do this.

**Comment by nene** [comment]
Sure. I also almost always use `.forEach()` instead. A pull request would be very welcome.

One interesting difference is that `.forEach()` also provides the loop index - with `for-of` you'll only have the element. Therefore the `for-of` transform currently refuses to transform loops that make use of loop index for other purposes than accessing the current element.

**Comment by aboyton** [comment]
Pull request added. I tried to reuse a bunch of the code from the `for-of` transformation, and rearranged the code a bit to do so.

I've also added as part of this the ability to do simple renaming of array item references. While it's not nearly as powerful as what's suggested in #169, I've found this makes transforming old a little easier. I haven't ported it to `for-of` loops as it's a little more difficult to do there.
